homeassistant/components/intellifire/coordinator.py

```diff
@@ -30,45 +30,32 @@ def __init__(
             name=DOMAIN,
             update_interval=timedelta(seconds=15),
         )
-        self._api = api
 
-    async def _async_update_data(self) -> IntellifirePollData:
-        if not self._api.is_polling_in_background:
-            LOGGER.info("Starting Intellifire Background Polling Loop")
-            await self._api.start_background_polling()
-
-            # Don't return uninitialized poll data
-            async with asyncio.timeout(15):
-                try:
-                    await self._api.poll()
-                except (ConnectionError, ClientConnectionError) as exception:
-                    raise UpdateFailed from exception
-
-        LOGGER.debug("Failure Count %d", self._api.failed_poll_attempts)
-        if self._api.failed_poll_attempts > 10:
-            LOGGER.debug("Too many polling errors - raising exception")
-            raise UpdateFailed
-
-        return self._api.data
+        self.fireplace = fireplace
 
     @property
-    def read_api(self) -> IntellifireAPILocal:
+    def read_api(self) -> IntelliFireDataProvider:
         """Return the Status API pointer."""
-        return self._api
+        return self.fireplace.read_api
 
     @property
-    def control_api(self) -> IntellifireAPILocal:
+    def control_api(self) -> IntelliFireController:
         """Return the control API."""
-        return self._api
+        return self.fireplace.control_api
+
+    async def _async_update_data(self) -> IntelliFirePollData:
+        return self.fireplace.data
```

Comment:
Does this force new data?

Reply:
It actually doesn't. The integration backing library runs its own update loop. This just returns the "latest version of the data we got from the background update loop"

Ultimately as I move towards `cloud`/`local` support (hopefully in the next PR) ... we will tell the backing lib which update loop it should be using and we'll just pull the latest copy from that data set.